src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -422,6 +422,13 @@ Error MeshForwarder::UpdateMeshRoute(Message &aMessage)
     mMeshDest      = meshHeader.GetDestination();
     mMeshSource    = meshHeader.GetSource();
 
+#if OPENTHREAD_CONFIG_MAC_DELAY_NEXT_TX_AFTER_FORWARDING_FRAME_TX
+    if (mMacDest.GetShort() != mMeshDest)
+    {
+        mDelayNextTx = true;
```

Comment:
seems MTD code never sets `mDelayNextTx` to true.
Does it mean MTD does not need mDelayNextTx logic?

Reply:
Yes. The current solution which is tied to inclusion of "lowpan mesh header" is only applicable to FTD (MTD won't include or parse mesh header).

There is still ongoing discussion in the Thread group and [JIRA](https://threadgroup.atlassian.net/browse/SPEC-1000) on the MTD behavior, whether we will have MTDs do anything  and if so what? (@simonlingoogle , please feel free to check the JIRA for this).  My personal preference is to not do add any such delay on MTD.

Reply:
If so, should we use #if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_DELAY_NEXT_TX_AFTER_FORWARDING_FRAME_TX in code like mesh_forwarder.cpp ?

Reply:
Yes. Good idea.
Let's keep this PR open now. We'll wait to see we converge on behavior under MTD this week.  If we limit this to FTD, will add the `#if OPENTHREAD_FTD` checks in the PR.

Reply:
Added `#if OPENTHREAD_FTD` to the check. The decision is to limit it to FTD (for now).
In the new push, renamed the feature to "collision avoidance delay" to reflect the changes from TC discussion.